scripts/setup_python311.sh

```
@@ -0,0 +1,118 @@
+# Determine venv name
+VENV_NAME="venv_py${PYTHON_VERSION}"
+PROJECT_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)"
+
+echo "Project root: $PROJECT_ROOT"
+echo "Virtual environment: $VENV_NAME"
+echo ""
+
+# Remove old venv if it exists
+if [ -d "$PROJECT_ROOT/$VENV_NAME" ]; then
+    echo -e "${YELLOW}Removing existing virtual environment...${NC}"
+    rm -rf "$PROJECT_ROOT/$VENV_NAME"
```

Shellcheck (0.11.0)

[warning] Use "${var:?}" to ensure this never expands to / .

(SC2115)